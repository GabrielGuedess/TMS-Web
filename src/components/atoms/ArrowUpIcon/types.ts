import { type ComponentPropsWithoutRef } from 'react';

export type ArrowUpIconProps = {
  size?: number;
} & ComponentPropsWithoutRef<'svg'>;
