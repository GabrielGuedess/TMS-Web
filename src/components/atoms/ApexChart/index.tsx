'use client';

import dynamic from 'next/dynamic';

import { type ApexChartProps } from './types';

const ReactApexChart = dynamic(() => import('react-apexcharts'), {
  ssr: false,
});

export const ApexChart = ({
  type,
  width,
  height,
  series,
  options,
}: ApexChartProps) => (
  <ReactApexChart
    type={type}
    width={width}
    series={series}
    height={height}
    options={options}
  />
);
