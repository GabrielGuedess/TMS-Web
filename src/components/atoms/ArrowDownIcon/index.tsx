import { forwardRef, type ForwardRefRenderFunction } from 'react';

import { type ArrowDownIconProps } from './types';

const ArrowDownIconRef: ForwardRefRenderFunction<
  SVGSVGElement,
  ArrowDownIconProps
> = ({ size = 10, ...props }, ref) => (
  <svg
    ref={ref}
    fill="none"
    width={size}
    height={size}
    viewBox="0 0 10 10"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      fill="currentColor"
      d="M5.64284 7.69237L9.09102 4.33987L10 5.22362L5 10.0849L-8.98488e-07 5.22362L0.908973 4.33987L4.35716 7.69237L4.35716 0.0848701L5.64284 0.0848704L5.64284 7.69237Z"
    />
  </svg>
);

export const ArrowDownIcon = forwardRef(ArrowDownIconRef);
