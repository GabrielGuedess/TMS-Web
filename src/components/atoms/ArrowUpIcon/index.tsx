import { forwardRef, type ForwardRefRenderFunction } from 'react';

import { type ArrowUpIconProps } from './types';

const ArrowUpIconRef: ForwardRefRenderFunction<
  SVGSVGElement,
  ArrowUpIconProps
> = ({ size = 10, ...props }, ref) => (
  <svg
    ref={ref}
    fill="none"
    width={size}
    height={size}
    viewBox="0 0 10 10"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      fill="currentColor"
      d="M4.35716 2.47737L0.908974 5.82987L5.0443e-07 4.94612L5 0.0848689L10 4.94612L9.09103 5.82987L5.64284 2.47737L5.64284 10.0849L4.35716 10.0849L4.35716 2.47737Z"
    />
  </svg>
);

export const ArrowUpIcon = forwardRef(ArrowUpIconRef);
