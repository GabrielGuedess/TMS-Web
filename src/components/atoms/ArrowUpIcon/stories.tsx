import { type Meta, type StoryFn } from '@storybook/react';

import { ArrowUpIcon } from '.';

import { type ArrowUpIconProps } from './types';

export default {
  args: { size: 24 },
  component: ArrowUpIcon,
  title: 'Atoms/ArrowUpIcon',
} as Meta<ArrowUpIconProps>;

export const Default: StoryFn<ArrowUpIconProps> = args => (
  <div className="flex w-full justify-center py-5 text-comet-500 dark:text-white-lilac-50">
    <ArrowUpIcon {...args} />
  </div>
);
