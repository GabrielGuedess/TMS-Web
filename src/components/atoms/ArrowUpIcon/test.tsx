import { render } from '@testing-library/react';

import { ArrowUpIcon } from '.';

describe('<ArrowUpIcon />', () => {
  it('should render the arrow up icon correctly', () => {
    const { container } = render(<ArrowUpIcon />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
