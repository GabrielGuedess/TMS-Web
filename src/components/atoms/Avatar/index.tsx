import Image from 'next/image';

import {
  forwardRef,
  type ElementRef,
  type ForwardRefRenderFunction,
} from 'react';

import { twMerge } from 'tailwind-merge';

import { type AvatarProps } from './types';

const AvatarRef: ForwardRefRenderFunction<
  ElementRef<typeof Image>,
  AvatarProps
> = ({ alt, src, className, ...props }, ref) => (
  <div
    className={twMerge(
      'relative w-10 h-10 select-none overflow-hidden rounded-full bg-porcelain-50 dark:bg-smoke-950 transition-all',
      className,
    )}
  >
    <Image
      ref={ref}
      src={src}
      alt={alt}
      className="select-none object-cover"
      sizes="(max-width: 768px) 100vw, 33vw"
      fill
      priority
      {...props}
    />
  </div>
);

export const Avatar = forwardRef(AvatarRef);
