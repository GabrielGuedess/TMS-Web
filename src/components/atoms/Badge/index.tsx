import { forwardRef, type ForwardRefRenderFunction } from 'react';

import { clsx } from 'clsx';
import { twMerge } from 'tailwind-merge';

import { type BadgeProps } from './types';

const BadgeRef: ForwardRefRenderFunction<HTMLDivElement, BadgeProps> = (
  { title, className, color = 'primary', variant = 'label', ...props },
  ref,
) => (
  <div className={twMerge('relative w-min', className)} {...props}>
    <div
      className={clsx('absolute left-0 top-0 size-full rounded', {
        'bg-shark-950': color === 'dark',
        'bg-cobalt-700': color === 'info',
        'bg-danger-500': color === 'danger',
        'opacity-[16%]': variant === 'label',
        'bg-primary-400': color === 'primary',
        'bg-success-500': color === 'success',
        'bg-warning-500': color === 'warning',

        'bg-dark-200 dark:bg-shark-950': color === 'secondary',
      })}
      ref={ref}
    />

    <div
      className={clsx('flex px-[0.625rem] py-[0.313rem]', {
        'text-white': variant === 'solid',

        'text-cobalt-700': color === 'info' && variant === 'label',
        'text-danger-500': color === 'danger' && variant === 'label',
        'text-primary-400': color === 'primary' && variant === 'label',
        'text-success-500': color === 'success' && variant === 'label',
        'text-warning-500': color === 'warning' && variant === 'label',
        'text-secondary-500': color === 'secondary' && variant === 'label',
        'bg-dark-200 dark:bg-shark-950':
          color === 'dark' && variant === 'label',
      })}
    >
      <span className="z-10 text-center text-[0.813rem] font-semibold capitalize">
        {title}
      </span>
    </div>
  </div>
);

export const Badge = forwardRef(BadgeRef);
