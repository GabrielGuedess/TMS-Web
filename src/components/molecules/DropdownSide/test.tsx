import { render } from '@testing-library/react';
import * as Accordion from '@radix-ui/react-accordion';

import { ArrowUpIcon } from 'components/atoms/ArrowUpIcon';

import { DropdownSide } from '.';

jest.mock('next/navigation', () => ({
  usePathname: () => '/',
}));

describe('<DropdownSide />', () => {
  it('should render the dropdown side correctly', () => {
    const { container } = render(
      <Accordion.Root type="single" collapsible>
        <DropdownSide subPath="/" title="Test" icon={ArrowUpIcon} isNavbarOpen>
          <h1>Test Children</h1>
        </DropdownSide>
        ,
      </Accordion.Root>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
