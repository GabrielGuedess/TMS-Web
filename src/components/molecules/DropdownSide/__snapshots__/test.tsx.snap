// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownSide /> should render the dropdown side correctly 1`] = `
<div
  data-orientation="vertical"
>
  <div
    class="min-w-min"
    data-orientation="vertical"
    data-state="closed"
  >
    <button
      aria-controls="radix-:r1:"
      aria-expanded="false"
      data-orientation="vertical"
      data-radix-collection-item=""
      data-state="closed"
      id="radix-:r0:"
      type="button"
    >
      <div
        class="flex w-auto items-center gap-9 rounded-xl p-3 outline-primary-400 transition-all text-comet-500 dark:text-dark-300 hover:bg-primary-400/90 hover:text-white-lilac-50 focus:bg-primary-400/90 focus:text-white-lilac-50 dark:hover:text-white-lilac-50 lg:w-64 lg:rounded lg:px-6 lg:py-2"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 10 10"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.35716 2.47737L0.908974 5.82987L5.0443e-07 4.94612L5 0.0848689L10 4.94612L9.09103 5.82987L5.64284 2.47737L5.64284 10.0849L4.35716 10.0849L4.35716 2.47737Z"
            fill="currentColor"
          />
        </svg>
        <div
          class="text-[0.938rem] font-light lg:flex"
        >
          <span>
            Test
          </span>
        </div>
      </div>
    </button>
    <div
      aria-labelledby="radix-:r0:"
      class="mt-4 flex min-w-min flex-col gap-2"
      data-orientation="vertical"
      data-state="closed"
      hidden=""
      id="radix-:r1:"
      role="region"
      style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
    />
  </div>
  ,
</div>
`;
